CurseForge publishing fails with getArchivePath() error

Tried to make a release for my mod via GitHub Actions, got a build error on CurseForge publishing. The package was built successfully, but it failed on publishing task with: `Could not find method getArchivePath() for arguments [] on task ':reobfJar' of type com.gtnewhorizons.retrofuturagradle.mcp.ReobfuscatedJar.`

Env: 
- RFG 2.0.2
- Gradle 9.2.1

CurseGradle calls getArchivePath() on the task passed to mainArtifact. This property was deprecated on AbstractArchiveTask in an earlier Gradle version and removed in Gradle 8. Since this template uses Gradle 9.2.1, the method no longer exists, causing a hard failure on the CurseForge publishing task.

Fix: Pass the resolved file directly instead of the task, bypassing CurseGradle's assumption entirely in `publishing.gradle`  

```
// Before
mainArtifact tasks.reobfJar, {

// After
mainArtifact tasks.reobfJar.archiveFile.get().asFile, {
```

<details>

<summary>Publishing log</summary>
<code>
  Gradle User Home already exists: will not restore from cache.
  /home/runner/work/ForkedEnderTweaker/ForkedEnderTweaker/gradlew curseforge
  WARNING: A restricted method in java.lang.System has been called
  WARNING: java.lang.System::load has been called by net.rubygrapefruit.platform.internal.NativeLibraryLoader in an unnamed module (file:/home/runner/.gradle/wrapper/dists/gradle-9.2.1-bin/2t0n5ozlw9xmuyvbp7dnzaxug/gradle-9.2.1/lib/native-platform-0.22-milestone-29.jar)
  WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
  WARNING: Restricted methods will be blocked in a future release unless native access is enabled
  

> Configure project :
Failed to check Java Version
> 
> ...
> Not relevant error about getConvention()
> org.gradle.internal.metaobject.AbstractDynamicObject$CustomMessageMissingMethodException: Could not find method getConvention() for arguments [] on root project 'ForkedEnderTweaker' of type org.gradle.api.Project
> ...
> 
> Task :compileInjectedInterfacesJava NO-SOURCE
> Task :extractDependencyATs UP-TO-DATE
> Task :generateForgeSrgMappings SKIPPED
> Task :downloadVanillaJars SKIPPED
> Task :mergeVanillaSidedJars SKIPPED
> Task :deobfuscateMergedJarToSrg SKIPPED
> Task :downloadFernflower SKIPPED
> Task :decompileSrgJar SKIPPED
> Task :cleanupDecompSrgJar SKIPPED
> Task :patchDecompiledJar SKIPPED
> Task :applyJST SKIPPED
> Task :remapDecompiledJar SKIPPED
> Task :decompressDecompiledSources UP-TO-DATE
> Task :processInjectedInterfacesResources NO-SOURCE
> Task :injectedInterfacesClasses UP-TO-DATE
> Task :compilePatchedMcJava UP-TO-DATE
> Task :processPatchedMcResources UP-TO-DATE
> Task :patchedMcClasses UP-TO-DATE
> Task :compileApiJava NO-SOURCE
> Task :processApiResources NO-SOURCE
> Task :apiClasses UP-TO-DATE
> Task :injectTags UP-TO-DATE
> Task :compileInjectedTagsJava UP-TO-DATE
> Task :extractNatives2 UP-TO-DATE
> Task :createMcLauncherFiles UP-TO-DATE
> Task :compileMcLauncherJava UP-TO-DATE
> Task :processInjectedTagsResources NO-SOURCE
> Task :injectedTagsClasses UP-TO-DATE
> Task :processMcLauncherResources NO-SOURCE
> Task :mcLauncherClasses UP-TO-DATE
> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :jar UP-TO-DATE
> Task :packageMcLauncher UP-TO-DATE
> Task :packagePatchedMc UP-TO-DATE
> Task :reobfJar UP-TO-DATE
> Task :sourcesJar UP-TO-DATE
> Task :assemble UP-TO-DATE
> Task :curseforge1475652 FAILED

[Incubating] Problems report is available at: file:///home/runner/work/ForkedEnderTweaker/ForkedEnderTweaker/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':curseforge1475652'.
> **Could not find method getArchivePath() for arguments [] on task ':reobfJar' of type com.gtnewhorizons.retrofuturagradle.mcp.ReobfuscatedJar.**

BUILD FAILED in 2s
</code>

</details>

--- a/gradle/scripts/publishing.gradle
+++ b/gradle/scripts/publishing.gradle
@@ -51,7 +51,7 @@ if (propertyBool('publish_to_curseforge')) {
                 changelog = parserChangelog()
                 changelogType = 'markdown'
             }
-            mainArtifact tasks.reobfJar, {
+            mainArtifact tasks.reobfJar.archiveFile.get().asFile, {
                 displayName = "${propertyString('mod_name')} ${propertyString('mod_version')}"
                 if (propertyBool('use_mixins')) {
                     relations {

